pkg/quota/evaluator/core/persistent_volume_claims.go

```diff
@@ -51,68 +54,127 @@ func listPersistentVolumeClaimsByNamespaceFuncUsingClient(kubeClient clientset.I
 	}
 }
 
+// ResourceByStorageClass returns a quota resource name by storage class.
+func ResourceByStorageClass(storageClass string, resourceName api.ResourceName) api.ResourceName {
+	return api.ResourceName(string(storageClass + storageClassSuffix + string(resourceName)))
+}
+
+// V1ResourceByStorageClass returns a quota resource name by storage class.
+func V1ResourceByStorageClass(storageClass string, resourceName v1.ResourceName) v1.ResourceName {
+	return v1.ResourceName(string(storageClass + storageClassSuffix + string(resourceName)))
+}
+
 // NewPersistentVolumeClaimEvaluator returns an evaluator that can evaluate persistent volume claims
 // if the specified shared informer factory is not nil, evaluator may use it to support listing functions.
 func NewPersistentVolumeClaimEvaluator(kubeClient clientset.Interface, f informers.SharedInformerFactory) quota.Evaluator {
-	allResources := []api.ResourceName{api.ResourcePersistentVolumeClaims, api.ResourceRequestsStorage}
+	fixedResources := []api.ResourceName{api.ResourcePersistentVolumeClaims, api.ResourceRequestsStorage}
+	usageFunc := PersistentVolumeClaimUsageFunc
+	constraintsFunc := makePersistentVolumeClaimConstraintsFunc(usageFunc)
 	listFuncByNamespace := listPersistentVolumeClaimsByNamespaceFuncUsingClient(kubeClient)
 	if f != nil {
 		listFuncByNamespace = generic.ListResourceUsingInformerFunc(f, schema.GroupResource{Resource: "persistentvolumeclaims"})
 	}
-
 	return &generic.GenericEvaluator{
```

Comment:
> I still think the registry.Store metaphor is fair.

`registry.Store` takes my input and provides 5 times the glue immediately and that glue integrates the functions I provide with higher level logic.  It's operating like an abstract class with final methods (as much as everyone hates have).

When first created, the `GenericEvaluator` took declarative values and did something similar.  As special cases arose, it has become a place to list functions which aren't glued together with higher level logic, but are largely just directly filled as impls.  Only `UsageStats` provides any value today and so I would reverse the links.